README.md

```diff
@@ -1,28 +1,65 @@
+## XMTP client SDK for pre-registration messaging
 
-**What is a memo?**
+`xmtp-memo-js` provides a TypeScript implementation of an XMTP memo client.
 
-Memo's are similar to XMTP Messages however they differ substantively.
+With XMTP, a blockchain account address must be registered on the XMTP network to send and receive end-to-end encrypted messages. Build your app with `xmtp-memo-js` to enable a registered address to start communicating with an unregistered address by sending a memo instead of a message.
 
-## Security Notice
+:warning: :warning: :warning: `XMTP:Memos` are available as an early Developer Preview. :warning: :warning: :warning: <br>
+The memo API will change without warning and may not be backward compatible.
 
-`xmtp-memo-js` uses [Lit protocol](https://developer.litprotocol.com/) to secure memos, and do not carry the same security characteristics as XMTP messages.
+### What is a memo?
 
-- Memos are **NOT** End-to-end encrypted. The encryption key is stored in the Lit Network and is only as secure as the escrow service.
+Memos are not the same as XMTP messages, though they are similar. For example, the payload structure of memos and messages is the same, but memos and messages are used in different circumstances.
+
+- Memos allow a registered address to send messages that are **not end-to-end encrypted** to **any Ethereum Virtual Machine-compatible blockchain address**.
+- Messages allow a registered address to send **end-to-end encrypted messages** to **any address registered on the XMTP network**.
+
+When possible, developers should always send messages with [XMTP-JS](https://github.com/xmtp/xmtp-js) to benefit from higher security and stability.
+
+### Memo sequence
+
+```mermaid
+sequenceDiagram
+autonumber
+    participant A as Amal
+    participant X as XMTP
+    participant B as Bola
+    A->>X: <Register>
+    Note left of A : Amal wants to message Bola
+    Note right of B : 2: Bola isn't on the network. <br> Can't message.
+    X -->> A: canMessage(Bola) : False
+    Note left of  A : 3, 4: Send memo instead
+    A->>X: Memo1: Hi Bola! It's Amal.
+    A->>X: Memo2: Msg me when you get this
+     Note over A,B: 1 month later
+    B->>X: <Register>
+
+    Note right of B : 6,7: Bola checks for memos
+    X -->>B: Memo1: Hi Bola! It's Amal.
+    X -->>B: Memo2: Msg me when you get this
+     Note right of B : 8: Send E2E encrypted message to Amal
+    B ->> X: Msg: Yo what's up Amal?
+    X -->> A: Msg: Yo what's up Amal?
+```
+
+## :warning: :warning: :warning: Security notice :warning: :warning: :warning:
+
+`xmtp-memo-js` uses [Lit protocol](https://developer.litprotocol.com/) to secure memos. Memos do not carry the same security characteristics as XMTP messages.
+
+- Memos are **NOT** end-to-end encrypted. The memo encryption key is stored in the Lit Network and is only as secure as the escrow service.
 - Memos have **NOT** undergone a formal security audit.
 
-## Installation
+## Install
```

Comment:
For the heading tweaks I made here and below, if you are interested, see the first row of the table here: https://developers.google.com/style/headings#heading-and-title-text

For doc sections that are about tasks - things someone needs to do - we use the "imperative mood." The form intends to make the title "action-oriented" - like a call-to-action on a UI button, for example - to help the user understand that they need to do something.

In the past, documentation used the gerund form (-ing) for task headings, but more recently the style has changed to move away from them. Tech writing trends! ;-)